pkg/integration/tests/branch/checkout_by_name.go

```diff
@@ -0,0 +1,38 @@
+package branch
+
+import (
+	"github.com/jesseduffield/lazygit/pkg/config"
+	. "github.com/jesseduffield/lazygit/pkg/integration/components"
+)
+
+var CheckoutByName = NewIntegrationTest(NewIntegrationTestArgs{
+	Description:  "Try to checkout branch by name.",
```

Comment:
Could we add some more to this description about the specific bug we're trying to defend against? That will help people understand why the test is slightly more complicated than just directly checking out a branch by name